components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaFetchRecords.java

```diff
@@ -343,6 +371,30 @@ protected void startPolling() {
             final KafkaRecordProcessorFacade recordProcessorFacade = createRecordProcessor();
 
             while (isKafkaConsumerRunnableAndNotStopped() && isConnected() && pollExceptionStrategy.canContinue()) {
+
+                if (commitRecordsRequested.compareAndSet(true, false)) {
+                    try {
+                        // we want to get details about last committed offsets (which MUST be done by this consumer thread)
+                        Map<TopicPartition, OffsetAndMetadata> commits = consumer.committed(consumer.assignment());
+                        commitRecords.clear();
+                        for (var e : commits.entrySet()) {
+                            KafkaTopicPosition p
+                                    = new KafkaTopicPosition(
+                                            e.getKey().topic(), e.getKey().partition(), e.getValue().offset(),
+                                            e.getValue().leaderEpoch().orElse(0));
+                            commitRecords.add(p);
+                        }
+                        CountDownLatch count = latch.get();
+                        if (count != null) {
+                            count.countDown();
+                        }
+                    } catch (Exception e) {
+                        // ignore cannot get last commit details
+                        LOG.debug("Cannot get last offset committed from Kafka brokers due to: {}. This exception is ignored.",
+                                e.getMessage(), e);
+                    }
+                }
+
```

Comment:
IMHO, it would be better to do this using a decorator. The biggest problem here is that we are attaching an edge case behavior right into the hot path of the component.

Reply:
No its only in use on special request

Reply:
> No its only in use on special request

So yeah, that's my point. We are calling `compareAndSet` here which, even though is an Intrinsic method, may actually cause a memory overhead due to need to perform CAS. 

So, even though an integration doesn't need the console, it would still be subject to this, which could hurt performance.

Reply:
And the reason I suggested a decorator, is because if we have a NOOP one and a Console one, then we could give the JVM a chance to use DCE to eliminate the code without hurting the performance.

Reply:
added a check if dev console enabled to skip the CAS